common/djangoapps/request_cache/__init__.py

```diff
@@ -26,3 +34,38 @@ def get_request():
     Return the current request.
     """
     return middleware.RequestCache.get_current_request()
+
+
+def get_request_or_stub():
+    """
+    Return the current request or a stub request.
+
+    If called outside the context of a request, construct a fake
+    request that can be used to build an absolute URI.
+
+    This is useful in cases where we need to pass in a request object
+    but don't have an active request (for example, in test cases).
```

Comment:
Why is this code in the file to be tested, rather than defined in a file that is explicitly used only by the tests? Having this function here is an invitation for other production code to call and use it.